node/lifecycle_test.go

```diff
@@ -103,13 +114,39 @@ func TestPodLifecycle(t *testing.T) {
 		}
 		t.Run("mockV0Provider", func(t *testing.T) {
 			assert.NilError(t, wireUpSystem(ctx, newMockV0Provider(), func(ctx context.Context, s *system) {
-				testCreateStartDeleteScenario(ctx, t, s)
+				testCreateStartDeleteScenario(ctx, t, s, waitForPodToBePermanentlyDeletedDeletionFunc)
 			}))
 		})
 	})
 
+	t.Run("createStartDeleteScenarioWithDeletionErrorNotFound", func(t *testing.T) {
+		// Wait for the pod to be started
+		mp := newMockProvider()
+		mp.errorOnDelete = errdefs.NotFound("not found")
+		assert.NilError(t, wireUpSystem(ctx, mp, func(ctx context.Context, s *system) {
+			testCreateStartDeleteScenario(ctx, t, s, waitForPodToBePermanentlyDeletedDeletionFunc)
+		}))
+	})
+
+	t.Run("createStartDeleteScenarioWithDeletionRandomError", func(t *testing.T) {
+		mp := newMockProvider()
+		// Wait for the pod to be started
+		deletionFunc := func(ev watch.Event) (bool, error) {
+			for atomic.LoadUint64(&mp.attemptedDeletes) < 2 {
```

Comment:
1.  This is a super hot loop, ok...
1.  If `attemptedDeletes` never reaches 2 we block forever?

Do these tests timeout?  A more informative test failure of "Didn't reach attempted deletes in X tries" is easier to debug later.

Reply:
the tests run with a timeout. Unfortunately, the timeout is a panic.

Reply:
🆒